src/wallet.ts

```diff
@@ -124,3 +125,52 @@ export class Wallet extends AdapterL2(AdapterL1(ethers.Wallet)) {
         return await this.provider.broadcastTransaction(await this.signTransaction(populatedTx));
     }
 }
+
+interface SigningFunction {
+    (this: AbstractWallet, transaction: TransactionRequest): Promise<string>;
+}
+
+export class AbstractWallet extends Wallet {
+    readonly accountAddress: string;
+    paymaster: Paymaster;
+    customSigningFunction: SigningFunction;
+
+    constructor(
+        accountAddress: string,
+        privateKey: string,
+        providerL2?: Provider,
+        providerL1?: ethers.Provider,
+        paymaster: Paymaster | null = null,
+        customSigningFunction: SigningFunction = signingFunction,
```

Comment:
Do we need a default here? Or should we just make it required?

Reply:
I find it makes the syntax for declaration simpler by adding a default here. I prefer it because of that. I can imagine some developers might not be interested in overriding the signing function and only seek to use AA for the custom logic you can add to accounts on-chain. In these cases I can imagine they'd be happy not having to be concerned with the signing logic, or paymaster logic either (if they do not wish to use a paymaster).